src/devices/bus/nes/mmc3_clones.cpp

```diff
@@ -640,20 +623,20 @@ void nes_nt639_device::pcb_reset()
 }
 
 // These reset-based PCBs are emulated here with no additional handlers
-void nes_resettxrom0_device::device_start()
+void nes_resettxrom_device::device_start()
 {
 	mmc3_start();
 	save_item(NAME(m_count));
 }
 
-void nes_resettxrom0_device::pcb_reset()
+void nes_resettxrom_device::pcb_reset()
 {
 	m_chr_source = m_vrom_chunks ? CHRROM : CHRRAM;
-	mmc3_common_initialize((1 << m_prg_shift) - 1, (1 << m_chr_shift) - 1, 0);
+	mmc3_common_initialize((m_outer_prg_size >> 3) - 1, m_outer_chr_size - 1, 0);
 
 	m_count = (m_count + 1) & 3;
-	m_prg_base = m_count << m_prg_shift;
-	m_chr_base = m_count << m_chr_shift;
+	m_prg_base = m_count << (m_outer_prg_size == 128 ? 4 : 5);  // << std::countr_zero(m_outer_prg_size) - 3; in C++20
+	m_chr_base = m_count << (m_outer_chr_size == 128 ? 7 : 8);  // << std::countr_zero(m_outer_chr_size); in C++20
```

Comment:
Can you not do this by counting leading zeroes rather than relying on fixed values?

Reply:
Changed to use `count_leading_zeros_32`. I had only poked around lib/util previously so wasn't aware of it.